test/nested-properties.test.js

```diff
@@ -0,0 +1,70 @@
+// Copyright IBM Corp. 2013,2016. All Rights Reserved.
+// Node module: loopback-datasource-juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// This test written in mocha+should.js
+'use strict';
+
+/* global getSchema:false, connectorCapabilities:false */
+var should = require('./init.js');
+
+var db, Model;
+
+describe('basic-querying', function() {
+  before(function(done) {
+    var modelDef = {
+      'nested': {
+        'duration': Number,
+        'happenedOn': Date,
+      },
+    };
+
+    db = getSchema();
+    Model = db.define('Model', modelDef);
+    db.automigrate(done);
+  });
+
+  describe('ping', function() {
+    it('should be able to test connections', function(done) {
+      db.ping(function(err) {
+        should.not.exist(err);
+        done();
+      });
+    });
+  });
+
+  describe('find', function() {
+    before(function(done) {
+      db = getSchema();
+      Model.destroyAll(done);
+    });
+
+    it('should query by nested Date: found', function(done) {
+      Model.create({
+        nested: {
+          duration: 2,
+          happenedOn: '2017-10-30T06:07:31.805Z',
+        },
+      }, function(err, m) {
+        should.not.exist(err);
```

Comment:
When this assertion fails, it hides the useful stack trace from `err` and shows useless stack trace pointing to this particular line instead. Here is the correct solution:

```js
if (err) return done(err);
```

(This won't be needed anymore after you switch to Promises, but I am still pointing it out for educational purposes.)